src/headers/session.rs

```diff
@@ -54,7 +54,7 @@ impl super::TypedHeader for Session {
             Some(header) => header,
         };
 
-        let mut iter = header.as_str().split(';');
+        let mut iter = header.as_str().split(";timeout=");
```

Comment:
While this will work for well-formed headers, it would probably be nicer to split on `;`, then iterate over that and find the first one that starts with `timeout`.

Some servers send random other fields and we probably shouldn't fail on that (the old code also would've failed on that of course).

Something like

```rust
        let mut iter = header.as_str().split(';');

        let session_id = iter.next().ok_or(HeaderParserError)?;

        let timeout = iter.find_map(|s| s.strip_prefix("timeout=").map(|s| s.parse::<u64>()).transpose().map_err(|_| HeaderParseError)?;
```

Reply:
I agree,  I will whip up some tests.

Reply:
Thanks!